Dart/src/com/jetbrains/lang/dart/ide/findUsages/DartServerFindUsagesHandlerFactory.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2000-2015 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.jetbrains.lang.dart.ide.findUsages;
+
+import com.intellij.find.findUsages.FindUsagesHandler;
+import com.intellij.find.findUsages.FindUsagesHandlerFactory;
+import com.intellij.psi.PsiElement;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class DartServerFindUsagesHandlerFactory extends FindUsagesHandlerFactory {
+  @Override
+  public boolean canFindUsages(@NotNull PsiElement element) {
+    return true;
```

Comment:
Here should be some smart filter. For example we shouldn't try to search for 'abstract' keyword.
Probably only DartNamedElement and DartUriElements are eligible, you should know better about server capabilities, what can it search for. Code injection in comments should somehow work too, but regular plain comment should not be searched for usages.